lib/elixir/src/elixir_module.erl

```diff
@@ -47,14 +47,11 @@ compile(Module, Block, Vars, #{line := Line} = Env) when is_atom(Module) ->
     _   -> Env#{lexical_tracker := nil, function := nil, module := Module}
   end,
 
-  case ?m(LexEnv, lexical_tracker) of
-    nil ->
-      elixir_lexical:run(?m(LexEnv, file), nil, fun(Pid) ->
-        do_compile(Line, Module, Block, Vars, LexEnv#{lexical_tracker := Pid})
-      end);
-    _ ->
-      do_compile(Line, Module, Block, Vars, LexEnv)
-  end;
+  Lexical = ?m(LexEnv, lexical_tracker),
+
+  elixir_lexical:run(?m(LexEnv, file), elixir_lexical:dest(Lexical), Lexical, fun(Pid) ->
+    do_compile(Line, Module, Block, Vars, LexEnv#{lexical_tracker := Pid})
+  end);
```

Comment:
This code now always creates a new lexical tracker for every module that gets compiled, with the current tracker as its parent.